HelpersWayland.cpp

```diff
@@ -0,0 +1,295 @@
+#ifdef WAYLAND_PROTOCOL
+
+#include <iostream>
+
+#include "HelpersWayland.h"
+
+bool HelpersWayland::hasProperLibEGL() {
+    driverName = (eglGetDisplayDriverName_t *) eglGetProcAddress("eglGetDisplayDriverName");
+    driverConfig = (eglGetDisplayDriverconfig_t *) eglGetProcAddress("eglGetDisplayDriverConfig");
+
+    if (!driverName || !driverConfig) {
+        std::cerr << "Error getting function pointers. LibEGL must be too old.\n";
+        return false;
+    }
+
+    //std::cout << "System has proper LibEGL\n";
+    return true;
+}
+
+
+void HelpersWayland::error_fatal(const char* format, ...) {
```

Comment:
This function has absolutely no relation to Wayland. Why has it been added here?
Also, why use such C-Like approach on almost pure C++ code?
If there is a fatal error, I prefer that an exception is throw.